services/search/pkg/search/service.go

```diff
@@ -72,10 +73,11 @@ func NewService(gatewaySelector pool.Selectable[gateway.GatewayAPIClient], eng e
 
 // Search processes a search request and passes it down to the engine.
 func (s *Service) Search(ctx context.Context, req *searchsvc.SearchRequest) (*searchsvc.SearchResponse, error) {
-	if req.Query == "" {
-		return nil, errtypes.BadRequest("empty query provided")
-	}
 	s.logger.Debug().Str("query", req.Query).Msg("performing a search")
+	err := resolveScope(req)
```

Comment:
How about changing this to something that doesn't mutate `req`, e.g. 
```
  ref, err := extractScope(req)
```
or something similar?

Reply:
done